patches/tModLoader/Terraria/ModLoader/ModNet.cs

```diff
@@ -262,18 +308,63 @@ internal static bool SyncClientMods(BinaryReader reader, out bool needsReload)
 			return false;
 		}
 
-		// ready to connect, apply configs. Config manager will apply the configs on reload automatically
+		// Even if needsReload, check configs anyway so user is more informed.
+		HashSet<string> modsWithConfigReload = new HashSet<string>();
+		foreach (NetConfig pendingConfig in pendingConfigs) {
+			if (modsWithConfigReload.Contains(pendingConfig.modname) || !ModLoader.HasMod(pendingConfig.modname))
+				continue;
+
+			ModConfig currentConfigClone = ConfigManager.GeneratePopulatedClone(ConfigManager.GetConfig(pendingConfig));
+			JsonConvert.PopulateObject(pendingConfig.json, currentConfigClone, ConfigManager.serializerSettingsCompact);
+			Mod mod = ModLoader.GetMod(pendingConfig.modname);
+			ModConfig loadTimeConfig = ConfigManager.GetLoadTimeConfig(mod, pendingConfig.configname);
+
+			if (loadTimeConfig.NeedsReload(currentConfigClone)) {
+				needsReload = true;
+				reloadRequiredExplanationEntries.Add(new ReloadRequiredExplanation(5, mod.Name, modFiles.FirstOrDefault(mod => mod.Name == pendingConfig.modname), $"[c/DDA0DD:Config change]"));
+				modsWithConfigReload.Add(pendingConfig.configname);
+				// We could mention the specific config if useful: \"{loadTimeConfig.DisplayName}\" Config change");
+				Logging.tML.Debug($"{pendingConfig.modname}:{pendingConfig.configname} ({loadTimeConfig.DisplayName}) would require a mod reload to join this server");
+			}
+		}
+
+		// If needsReload is still false, apply configs to real ModConfig instances and join server directly
 		if (!needsReload) {
 			foreach (NetConfig pendingConfig in pendingConfigs)
 				JsonConvert.PopulateObject(pendingConfig.json, ConfigManager.GetConfig(pendingConfig), ConfigManager.serializerSettingsCompact);
```

Comment:
Is this common with above code?

Reply:
No, it needs to be separate, one populates a clone for the check, the other populates the active config directly.